models/components.go

```diff
@@ -0,0 +1,539 @@
+package models
+
+import (
+	"context"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/flanksource/commons/logger"
+	"github.com/flanksource/duty/types"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/schema"
+)
+
+type Components []*Component
+
+type Component struct {
+	Checks           Checks                   `json:"checks,omitempty" gorm:"-"`
+	ComponentChecks  ComponentChecks          `json:"-" gorm:"column:component_checks" swaggerignore:"true"`
+	Components       Components               `json:"components,omitempty" gorm:"-"`
+	ConfigInsights   []map[string]interface{} `json:"insights,omitempty" gorm:"-"`
+	Configs          Configs                  `json:"configs,omitempty" gorm:"type:configs"`
+	CostPerMinute    float64                  `json:"cost_per_minute,omitempty" gorm:"column:cost_per_minute"`
+	CostTotal1d      float64                  `json:"cost_total_1d,omitempty" gorm:"column:cost_total_1d"`
+	CostTotal30d     float64                  `json:"cost_total_30d,omitempty" gorm:"column:cost_total_30d"`
+	CostTotal7d      float64                  `json:"cost_total_7d,omitempty" gorm:"column:cost_total_7d"`
+	CreatedAt        time.Time                `json:"created_at,omitempty" time_format:"postgres_timestamp"`
+	DeletedAt        *time.Time               `json:"deleted_at,omitempty" time_format:"postgres_timestamp" swaggerignore:"true"`
+	ExternalId       string                   `json:"external_id,omitempty"` //nolint
+	Icon             string                   `json:"icon,omitempty"`
+	ID               uuid.UUID                `json:"id,omitempty" gorm:"default:generate_ulid()"` //nolint
```

Comment:
Keep ID as the first, looks better

Reply:
okay. I generally sort fields in a relatively large struct so it's easier to scan